src/Model/CachingIterator.php

```diff
@@ -40,6 +40,9 @@
  */
 class CachingIterator implements Countable, Iterator
 {
+    private const FIELD_KEY = 'key';
+    private const FIELD_VALUE = 'value';
```

Comment:
I'd suggest defining `FIELD_KEY` as `0` and `FIELD_VALUE` as `1`. This will allow PHP to use a packed array internally and avoid the overhead of storing key strings for each cached element.

Alternatively, we can do without the constants and just treat `$items` as a tuple assume the key and value are the first and second elements of each item, respectively. That would entail the following change in `storeCurrentItem()`:

```diff
- $this->items[$this->iterator->key()] = $this->iterator->current();
+ $this->items[] = [$this->iterator->key(), $this->iterator->current()];
```

I'll defer to you on whether you prefer to use constants (it's a small class so either approach is fine), but let's definitely avoid the strings.